### PR TITLE
Define NumericType in client.py alaways the NumericType = typing.Union[int, float]

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

When running from a new device, it always reports `NumericType` not found. This might because `typing.TYPE_CHECKING` is false, in this case we either always define `NumericType` to be the typing union or we define an else.  

### Why are the changes needed?

Make onboarding easier.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing UT

### CODE REVIEW

**Comment by amaliujia** [comment]
R: @grundprinzip

File: python/pyspark/sql/connect/client.py [hunk 1]
```diff
@@ -32,8 +32,9 @@
 from pyspark.sql.connect.data_frame import DataFrame
 from pyspark.sql.connect.plan import Read, Sql
 
-if typing.TYPE_CHECKING:
-    NumericType = typing.Union[int, float]
+
+NumericType = typing.Union[int, float]
```
Comment (amaliujia) [inline python/pyspark/sql/connect/client.py:35]: or we should define a `else`. I am not familiar with Python and not sure how to enable `typing.TYPE_CHECKING`. so I hit `NumericType` not found

Comment (amaliujia) [reply python/pyspark/sql/connect/client.py:35]: Another option is if there is a way to configure typing.TYPE_CHECKING=true as default.

Comment (grundprinzip) [reply python/pyspark/sql/connect/client.py:35]: Good catch. I think for whatever reason I didn't check this properly. The `TYPE_CHECKING` global is only set when mypy is doing type checking. TBH the whole python type checking is just for editors but there is no runtime support really.

**Review by grundprinzip — APPROVED** [review]

**Pull request merged**